File: data/src/main/java/com/br/natanbrito/challenge/data/model/results/Address.kt
package com.br.natanbrito.challenge.data.model.results

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Address(
    val city: String,
    val country: String,
    val geoLocation: GeoLocation,
    @SerializedName("id_atlas_city")
    val idAtlasCity: Int?,
    @SerializedName("id_atlas_country")
    val idAtlasCountry: Int?,
    @SerializedName("id_atlas_neighborhood")
    val idAtlasNeighborhood: Int?,
    @SerializedName("id_atlas_state")
    val idAtlasState: Int?,
    @SerializedName("id_city")
    val idCity: Int,
    @SerializedName("id_country")
    val idCountry: Int,
    @SerializedName("id_state")
    val idState: Int,
    val state: String,
    val street: String,
    val zipcode: String
) : Parcelable

@Parcelize
data class GeoLocation(
    val lat: Double,
    val lon: Double
) : Parcelable

File: Hurb_iOS_Challenge/Hurb_iOS_Challenge/Model/Address.swift
//
//  Address.swift
//  Hurb_iOS_Challenge
//
//  Created by Gáudio Ney on 22/05/22.
//

import Foundation

// MARK: - Address
struct Address: Codable {
    let city: City?
    let country: Country?
    let idAtlasCity, idAtlasCountry, idAtlasNeighborhood, idAtlasState: Int?
    let idCity, idCountry, idState: Int?
    let state: State?
    let street, zipcode: String?
    let geoLocation: GeoLocation?

    enum CodingKeys: String, CodingKey {
        case city, country
        case idAtlasCity = "id_atlas_city"
        case idAtlasCountry = "id_atlas_country"
        case idAtlasNeighborhood = "id_atlas_neighborhood"
        case idAtlasState = "id_atlas_state"
        case idCity = "id_city"
        case idCountry = "id_country"
        case idState = "id_state"
        case state, street, zipcode, geoLocation
    }
}
